crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -1583,139 +1593,144 @@ impl TryFrom<StoredTransaction> for TransactionBlock {
 
-        let validator_set = ValidatorSet {
-            total_stake: e.new_total_stake.map(|s| BigInt::from(s as u64)),
-            active_validators: Some(active_validators),
-            ..Default::default()
-        };
+    let validator_set = ValidatorSet {
+        total_stake: e.new_total_stake.map(|s| BigInt::from(s as u64)),
+        active_validators: Some(active_validators),
+        ..Default::default()
+    };
 
-        let net_inflow =
-            if let (Some(fund_inflow), Some(fund_outflow)) = (e.storage_charge, e.storage_rebate) {
-                Some(BigInt::from(fund_inflow - fund_outflow))
-            } else {
-                None
-            };
+    let net_inflow =
+        if let (Some(fund_inflow), Some(fund_outflow)) = (e.storage_charge, e.storage_rebate) {
+            Some(BigInt::from(fund_inflow - fund_outflow))
+        } else {
+            None
+        };
 
-        Ok(Self {
-            epoch_id: e.epoch as u64,
-            protocol_version: e.protocol_version as u64,
-            reference_gas_price: Some(BigInt::from(e.reference_gas_price as u64)),
-            validator_set: Some(validator_set),
-            start_timestamp: DateTime::from_ms(e.epoch_start_timestamp),
-            end_timestamp: e.epoch_end_timestamp.and_then(DateTime::from_ms),
-            total_checkpoints: e
-                .last_checkpoint_id
-                .map(|last_chckp_id| BigInt::from(last_chckp_id - e.first_checkpoint_id)),
-            total_gas_fees: e.total_gas_fees.map(BigInt::from),
-            total_stake_rewards: e.total_stake_rewards_distributed.map(BigInt::from),
-            total_stake_subsidies: e.stake_subsidy_amount.map(BigInt::from),
-            fund_size: e.storage_fund_balance.map(BigInt::from),
-            net_inflow,
-            fund_inflow: e.storage_charge.map(BigInt::from),
-            fund_outflow: e.storage_rebate.map(BigInt::from),
-        })
-    }
+    Ok(Epoch {
+        epoch_id: e.epoch as u64,
+        protocol_version: e.protocol_version as u64,
+        reference_gas_price: Some(BigInt::from(e.reference_gas_price as u64)),
+        validator_set: Some(validator_set),
+        start_timestamp: DateTime::from_ms(e.epoch_start_timestamp),
+        end_timestamp: e.epoch_end_timestamp.and_then(DateTime::from_ms),
+        total_checkpoints: e
+            .last_checkpoint_id
+            .map(|last_chckp_id| BigInt::from(last_chckp_id - e.first_checkpoint_id)),
+        total_gas_fees: e.total_gas_fees.map(BigInt::from),
+        total_stake_rewards: e.total_stake_rewards_distributed.map(BigInt::from),
+        total_stake_subsidies: e.stake_subsidy_amount.map(BigInt::from),
+        fund_size: e.storage_fund_balance.map(BigInt::from),
+        net_inflow,
+        fund_inflow: e.storage_charge.map(BigInt::from),
+        fund_outflow: e.storage_rebate.map(BigInt::from),
+    })
 }
 
-impl TryFrom<NativeSuiSystemStateSummary> for SuiSystemStateSummary {
```

Comment:
This `TryFrom` is replaced with a free function as we need to pass `validators_apy` data to create the `active validators`.